java/core/src/java/org/apache/orc/OrcFile.java

```diff
@@ -350,18 +346,6 @@ public KeyProvider getKeyProvider() {
       return keyProvider;
     }
 
-    /**
-     * @deprecated Use {@link #orcTail(OrcTail)} instead.
-     */
-    public ReaderOptions fileMetadata(final FileMetadata metadata) {
-      fileMetadata = metadata;
-      return this;
-    }
-
-    public FileMetadata getFileMetadata() {
```

Comment:
As @mystic-lama pointed, Apache ORC community cannot drop the public API without a deprecation, @deshanxiao .

Reply:
+1 for @mystic-lama points. Acutally this PR targets 2.0

Reply:
@deshanxiao What I meant is we need one more PR with this API marked as deprecated in a version that goes out before 2.0.0, so that people can expect the change. I don't have strong opinion on this, given breaking changes are expected in major release. Please choose what is least disruptive. Once we make a decision on this, I can give my +1

A big Thank You for deleting dead code :bowing_man:

Reply:
There are two more comments.
1. +1 for another independent deprecation PR for specific discussion about that.
2. Please note that `Deperecated Code` is not a dead code. If there is any usage in the downstream, we may not able to delete the deprecated code. For example, we recovered the deleted code back.

ORC-677: Add a deprecated legacy constructor SargApplier back (#557)
ORC-676: Add getRawDataSizeFromColIndices back to ReaderImpl (#555)
ORC-671: Add OrcTail.getStripeStatistics back for backward compatiblility (#549)

We had better be very careful even in 2.0.0. Semantic Versioning means it allows breaking changes, but doesn't mean we need to break absolutely. The fundamental purpose of `Semantic Versioning` is to help users.

Reply:
Thank you @dongjoon-hyun @mystic-lama . I will open an independent PR.